src/extendedstorage.cpp

```diff
@@ -123,11 +107,46 @@ bool ExtendedStorage::Private::clear()
 
     return true;
 }
+
+Incidence::List ExtendedStorage::Private::incidencesWithAlarms(const QString &notebookUid, const QString &uid)
+{
+    Incidence::List list;
+    if (!mNotebooks.contains(notebookUid)
+        || !mNotebooks.value(notebookUid)->isVisible()) {
+        return list;
+    }
+
+    if (uid.isEmpty()) {
+        Incidence::List all;
+        mStorage->allIncidences(&all, notebookUid);
```

Comment:
Guess this is ok since this branch should happen only when notebook becomes visible, like it was getting all the instances earlier too. But noticed there's a slight inconsistency when this branch loads the related incidences while the other one fetches the the parent item from calendar, omitting loading.

So in other words, this is kind of making assumptions on which kind of context the alarm setup or incidence fetching are done. Maybe still ok with the current usage, but a bit unfortunate, I guess. Maybe some note here could be in place?

Reply:
That's a good remark. I've added a long documentation in AlarmHandler::incidencesWithAlarms() to explain when this function is called and what can be the assumption on incidence availability in memory at the time of call.

I've summurized it also in extendedstorage to be clear.